### DISCUSSION TITLE
HEIC image format readable?

### DISCUSSION THREAD

**Discussion by toysonlocation** [airsdk/Adobe-Runtime-Support#1679] [Ideas]
Anyone know of a library than can, or an SDK plan to, load HEIC images?

**Comment by marchbold** [comment]
Our camera roll extension will load heic when a user selects it but that's not really a direct way to load a heic file. Are you looking for a cross platform solution or iOS only?

**Reply by toysonlocation** [reply]
I left out a crucial piece of info: this is for macOS. I would like to be able to do this to files/bytearrays, but if at-file-selection-time is possible, on macOS, I’d take that for now.

Also to clarify, by "load" I mean put into the display list, as a Bitmap.

**Reply by ajwfrost** [reply]
My main concern with HEIC is the fact there may be patent licensing requirements, depending on how you're using it... if you could use the native macOS tools to convert the image to png/jpeg then it would be better I think!

**Reply by toysonlocation** [reply]
Yes, I was fearing I may need to dip into the world of ANE and mine the bits myself. Can you suggest a "best" code example for making macOS ANEs? Thanks!

**Reply by marchbold** [reply]
Shouldn't be too hard, I can throw something together for you if you want.

**Reply by toysonlocation** [reply]
Well I would love that for sure. If you do, I'll donate a BIG coffee to the project author.
